### ISSUE TITLE
babel-plugin-transform-es2015-modules-amd produces invalid AMD modules

### ISSUE THREAD

**Issue by jslegers** [babel/babel#9791]
## Bug Report

I have the following `package.json` :

````json
{
  "name": "watchman",
  "version": "1.0.0",
  "description": "Simple watcher for ES6 to AMD conversion",
  "author": "Luciad NV",
  "license": "MIT",
  "scripts": {
    "build": "cross-env NODE_OPTIONS=--max_old_space_size=8192 babel es6/geometry es6/ria es6/symbology -d release --watch"
  },
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-plugin-transform-es2015-modules-amd": "^6.24.1",
    "babel-preset-env": "^1.7.0",
    "cross-env": "^5.2.0"
  }
}
````
 
I have the following `.babelrc` :

````json
{
  "presets": ["babel-preset-env"],
  "plugins": ["transform-es2015-modules-amd"]
}
````
I have the following es6 module in my `es6/ria` folder :

````js
import Object from "./Dependency";

  export default {
   
  };
````

If I run `npm run build` This produces an AMD module with the following implementation :

````js
define(["exports", "./Dependency"], function (exports, _Dependency) {
  "use strict";

  _Dependency2.default.defineProperty(exports, "__esModule", {
    value: true
  });

  var _Dependency2 = _interopRequireDefault(_Dependency);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  exports.default = {};
});
````

As you can see, `_Dependency2` is defined only after `_Dependency2.default` is referenced, and this throws an error.

**Comment by nicolo-ribaudo** [comment]
This is probably because you are transpiling both to commonjs (using the preset) and to amd. Could you try this configuration?
```json
{
  "presets": ["babel-preset-env", { "modules" : "amd" }]
}
```

PS. Babel 6 isn't maintained anymore, I suggest using Babel 7.

**Comment by jslegers** [comment]
> Could you try this configuration?

Your suggested change actually made things worse.
The first time, at least output was produced.
Now, I get a compile-time ReferenceError with the following stack trace :

```
ReferenceError: [BABEL] es6\ria\Evented.js: Using removed Babel 5 option: foreign.modules - Use the corresponding module transform plugin in the `plugins` option. Check out http://babeljs.io/docs/plugins/#modules
    at Logger.error (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\logger.js:41:11)
    at OptionManager.mergeOptions (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:220:20)
    at D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:265:14
    at D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:323:22
    at Array.map (<anonymous>)
    at OptionManager.resolvePresets (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:275:20)
    at OptionManager.mergePresets (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:264:10)
    at OptionManager.mergeOptions (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:249:14)
    at OptionManager.init (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\options\option-manager.js:368:12)
    at File.initOptions (D:\Desktop\test-es6\watchman-test-2\node_modules\babel-core\lib\transformation\file\index.js:212:65)
```
__________________

> Babel 6 isn't maintained anymore, I suggest using Babel 7.

I already have. 
And I replaced this plugin with `@babel/plugin-transform-modules-amd`.
That fixed this issue.

Considering Babel 6 isn't maintained anymore, I suppose this ticket can be closed?

Anyway, there's still one another issue that is blocking my progress. 
I created another ticket for that (which you also already addressed).
See #9793.

**Comment by nicolo-ribaudo** [comment]
Sorry, I forgot some parentheses:
```json
{
  "presets": [
    ["@babel/preset-env", { "modules" : "amd" }]
  ]
}
```

> Considering Babel 6 isn't maintained anymore, I suppose this ticket can be closed?

I think that you would have the same problem with Babel 7, but the config above should fix it.
